packages/core/components/DropZone/index.tsx

```diff
@@ -13,12 +13,19 @@ const getClassName = getClassNameFactory("DropZone", styles);
 
 export { DropZoneProvider, dropZoneContext } from "./context";
 
-type DropZoneProps = {
+type DropZoneProps<Props extends Record<string, any> = Record<string, any>> = {
```

Comment:
Can we use a different shape and name for the generic type here?

```tsx
type DropZoneProps<ValidComponents extends string[] = string[]> = {
```

Then folk can instantiate with `keyof Config.components`

Reply:
I was aiming to match the type of the props-mapping-type (for lack of a better term). So you can reuse the same type here as you do when defining your config.

We could make it instead:

```tsx
type DropZoneProps<T extends string> = {...etc};
```

And then you can provide the following to reuse the existing types:

```tsx
<DropZone<keyof Props> />
```

Reply:
Yup I think that makes sense